### ISSUE TITLE
Scala Console's `:javap` functionality 

### ISSUE THREAD

**Issue by spockz** [lihaoyi/Ammonite#272]
When giving courses I switched to the Ammonite REPL because of the syntax-highlighting and other nice features. The one thing I'm missing is the possibility to show the Java bytecode for classes that were created during the session as I can do with the Scala Console's `:javap`.

```scala
scala> class Foo
defined class Foo
scala> :javap -c Foo
Compiled from "<console>"
public class Foo {
  public Foo();
    Code:
       0: aload_0
       1: invokespecial #9                  // Method java/lang/Object."<init>":()V
       4: return
}
```

As suggested by @lihaoyi and @jrudolph , this could be an approach:
>lihaoyi 11:22
>if i were planning the project
>it'd go something like
>expose a getClassFile[MyClass] function in the REPL
>Write/find/steal a bytecode pretty-printer
>Combine with a helper function called javap[MyClass] and profit

>lihaoyi 11:24
number 1. shouldn't be too hard because you can just ask for the bytes of a class via getResourceAsStream
number 2. I have no clue, worse come to worst you write your own

>jrudolph 11:29
>number 2. => e.g. http://asm.ow2.org/asm50/javadoc/user/org/objectweb/asm/util/Textifier.html

**Comment by jrudolph** [comment]
`getResourceAsStream` won't work for classes defined in the REPL itself because ammonite's SpecialClassLoader doesn't implement it. I guess it would make sense to use `SpecialClassLoader.newFileDict` directly to access bytecode for those classes.

**Comment by lihaoyi** [comment]
Yeah, it wouldn't work directly, but fixing `SpecialClassLoader` to do the right thing wouldn't be too hard. Of course we could also go to `newFileDict` directly =)

Only the minimal changes to expose the classfile bytes should live in `ammonite-repl`, everything else should probably live in a separate package called `ammonite-scalac` or something

**Comment by lihaoyi** [comment]
Closed in deference to #286
